proto/desmos/profiles/v1beta1/models.proto

```diff
@@ -125,24 +127,46 @@ message Proof {
 
 // ChainConfig contains the data of the chain with which the link is made.
 message ChainConfig {
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = true;
+
   string name = 1
       [ (gogoproto.moretags) = "yaml:\"name\"" ];
-  string bech32_addr_prefix = 2
-      [ (gogoproto.moretags) = "yaml:\"bech32_addr_prefix\"" ];
 }
 
 // ChainLink contains the data representing either an inter- or cross- chain link
 message ChainLink {
-  string address = 1 [ (gogoproto.moretags) = "yaml:\"address\"" ];
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = true;
+  
+  google.protobuf.Any address = 1 [ (gogoproto.moretags) = "yaml:\"address\"" ];
   Proof proof = 2
       [ (gogoproto.nullable) = false, (gogoproto.moretags) = "yaml:\"proof\"" ];
   ChainConfig chain_config = 3 [
     (gogoproto.nullable) = false,
     (gogoproto.moretags) = "yaml:\"chain_config\""
   ];
-  google.protobuf.Timestamp creation_time = 4 [
+
+  google.protobuf.Timestamp creation_time = 5 [
     (gogoproto.stdtime) = true,
     (gogoproto.moretags) = "yaml:\"creation_time\"",
     (gogoproto.nullable) = false
   ];
 }
+
+ // Bech32Address represents a Bech32-encoded address
+ message Bech32Address {
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = true;
+
```

Comment:
Can we add the `cosmos_proto.implements_interface` Protobuf option here the same way that [Cosmos does](https://github.com/cosmos/cosmos-sdk/blob/master/proto/cosmos/auth/v1beta1/auth.proto#L18? I think it will be helpful in the future:

```protobuf
option (cosmos_proto.implements_interface) = "AddressData";
```